reco_utils/recommender/newsrec/IO/news_iterator.py

```diff
@@ -0,0 +1,308 @@
+            elif "CandidateNews" in tokens[0]:
+                # word index start by 0
+                for i in tokens[1].split(","):
+                    candidate_news_index.append(int(i))
+            elif "ClickedNews" in tokens[0]:
+                click_news_index.append([int(i) for i in tokens[1].split(",")])
+            else:
+                raise ValueError("data format is wrong")
+
+        return (
+            label,
+            imp_index,
+            user_index,
+            candidate_news_index,
+            click_news_index
+        )
+
+
+    def load_data_from_file(self, infile):
+        """Read and parse data from a file.
+        
+        Args:
+            infile (str): text input file. Each line in this file is an instance.
+
+        Returns:
+            obj: An iterator that will yields parsed results, in the format of graph feed_dict.
+        """
+        label_list = []
+        imp_indexes = []
+        user_indexes = []
+        candidate_news_indexes = []
+        click_news_indexes = []
+        cnt = 0
+
+        with tf.gfile.GFile(infile, "r") as rd:
+            while True:
+                line = rd.readline()
+                if not line:
+                    break
+
+                label, imp_index, user_index, candidate_news_index, click_news_index= self.parser_one_line(line)
+
+                candidate_news_indexes.append(candidate_news_index)
+                click_news_indexes.append(click_news_index)
+                label_list.append(label)
+                imp_indexes.append(imp_index)
+                user_indexes.append(user_index)
+
+                cnt += 1
+                if cnt >= self.batch_size:
+                    yield self._convert_data(
+                        label_list, 
+                        imp_indexes,
+                        user_indexes,
+                        candidate_news_indexes, 
+                        click_news_indexes)
+                    candidate_news_indexes = []
+                    click_news_indexes = []
+                    label_list = []
+                    imp_indexes = []
+                    user_indexes = []
+                    cnt = 0
+
+    def _convert_data(
+        self,
+        label_list,
+        imp_indexes,
+        user_indexes,
+        candidate_news_indexes,
+        click_news_indexes
+    ):
+        """Convert data into numpy arrays that are good for further model operation.
+        
+        Args:
+            label_list (list): a list of ground-truth labels.
+            imp_indexes (list): a list of impression indexes.
+            user_indexes (list): a list of user indexes.
+            candidate_news_indexes (list): the candidate news article's words indices
+            click_news_indexes (list): words indices for user's clicked news articles
+            
+        Returns:
+            dict: A dictionary, contains multiple numpy arrays that are convenient for further operation.
```

Comment:
the return value is a list, should it be a dict instead? that would certainly make it easier to extract results without needing to know the order of the output.